clients/src/main/java/org/apache/kafka/common/requests/MetadataResponse.java

```diff
@@ -211,20 +236,53 @@ public Cluster cluster() {
         return brokers;
     }
 
+    /**
+     * Get all topic metadata returned in the metadata response
+     * @return the topicMetadata
+     */
+    public Collection<TopicMetadata> topicMetadata() {
+        return topicMetadata;
+    }
+
+    /**
+     * Get all the controller node returned in metadata response
+     * @return the controller node or null if it doesn't exist
+     */
+    public Node controller() {
+        Node controller = null;
+        for (Node broker : brokers) {
+            if (broker.isController()) {
+                controller = broker;
+                break;
+            }
+        }
+        return controller;
+    }
+
     public static MetadataResponse parse(ByteBuffer buffer) {
-        return new MetadataResponse(CURRENT_SCHEMA.read(buffer));
+        return parse(buffer, CURRENT_VERSION);
+    }
+
+    public static MetadataResponse parse(ByteBuffer buffer, int version) {
+        return new MetadataResponse(ProtoUtils.responseSchema(ApiKeys.METADATA.id, version).read(buffer));
     }
 
     public static class TopicMetadata {
         private final Errors error;
         private final String topic;
+        private final boolean isInternal;
+        private final boolean markedForDeletion;
```

Comment:
Yeah, that could save a byte for each topic. I will mention it during the wire protocol discussion on the next kip call, since this needs to be reviewed/voted.